### PR TITLE
Doc translation (French): Concepts.

### PR DESCRIPTION
**Pull request by GitGud31**
Attempt for **Documentation translation** (French).

Section: **Concepts**
1.  reading.mdx
2.  provider_observer.mdx
3.  combining_providers.mdx

### CODE REVIEW

**Comment by GitGud31** [comment]
The second commit is just an edit of a word after doing a revision. 

Please let me know if something isn't clear, or there are still some typos and such.

Cordially,

File: website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/provider_observer.mdx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+- `didAddProvider` est appelé chaque fois qu'un provider a été initialisé, et la valeur exposée est value.
+- `didDisposeProvider` est appelé chaque fois que le provider A a été éliminé.
+- `didUpdateProvider` est appelé à chaque fois que mes providers émettent une notification.
+- `mayHaveChanged` est appelé lorsque la dépendance d'un provider a changé, mais il n'est pas encore certain que la valeur calculée change.
+
+:::info
+Il est possible que mayHaveChanged soit appelé, sans que didUpdateProvider soit appelé, 
+par exemple lorsqu'un Provider est recalculé mais renvoie une valeur == à la précédente.
+:::
```
Comment (rrousselGit) [inline website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/provider_observer.mdx:18]:
```suggestion
```

Comment (rrousselGit) [reply website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/provider_observer.mdx]: This life-cycle was removed. The en doc is out of date

Comment (GitGud31) [reply website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/provider_observer.mdx]: Sorry, I didn't pay attention to changes, I was focused more on the translation.

I will update (remove 'hayHaveChanged' part) in the English documentation as well.

Other than that is the french translation for the **Concepts** section all good? So I can continue doing the rest of the doc.

**Comment by GitGud31** [comment]
Just an update of the outdated english doc

File: website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/combining_providers.mdx [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+```
+
+Si vous avez utilisé [read] pour tenter d'éviter les reconstructions non souhaitées de votre objet, reportez-vous à 
+[Mon provider fait des mises à jour trop souvent, que puis-je faire ?](#my-provider-updates-too-often-what-can-i-do)
+Mauvaise pratique pour appeler `read` ici
+:::
+
+### Comment tester un objet qui reçoit [read] comme paramètre de son constructeur ?
+
+Si vous utilisez le modèle décrit dans [Puis-je lire un provider sans l'écouter ?](#can-i-read-a-provider-without-listening-to-it),
```
Comment (rrousselGit) [inline website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/combining_providers.mdx:201]: Those links need to be updated to use the French description

Comment (rrousselGit) [reply website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/combining_providers.mdx:201]: There are a few of those, so they should all be edited :)

**Comment by rrousselGit** [comment]
Thanks!

Just a few links to fix and it'll be good to merge 😄

File: website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/combining_providers.mdx (hunk 2 above)
Comment (GitGud31) [reply website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/combining_providers.mdx:201]: I'll go through all of them again, no problem 👍

**Comment by GitGud31** [comment]
All links should be in match with French description. Hopefully 😅

File: website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/combining_providers.mdx [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+```
+
+Si vous avez utilisé [read] pour tenter d'éviter les reconstructions non souhaitées de votre objet, reportez-vous à 
+[Mon provider fait des mises à jour trop souvent, que puis-je faire ?](#Mon-provider-fait-des-mises-à-jour-trop-souvent,-que-puis-je-faire-?)
+Mauvaise pratique pour appeler `read` ici
+:::
+
+### Comment tester un objet qui reçoit [read] comme paramètre de son constructeur ?
+
+Si vous utilisez le modèle décrit dans [Peut-on lire un provider sans l'écouter ?](#Peut-on-lire-un-provider-sans-l'écouter-?),
```
Comment (rrousselGit) [inline website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/combining_providers.mdx:201]:
```suggestion
Si vous utilisez le modèle décrit dans [Peut-on lire un provider sans l'écouter ?](#peut-on-lire-un-provider-sans-lécouter-), 
```

File: website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/combining_providers.mdx [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+```dart
+final myProvider = Provider((ref) {
+  // Mauvaise pratique d'appeler `read` ici.
+  final value = ref.read(anotherProvider);
+});
+```
+
+Si vous avez utilisé [read] pour tenter d'éviter les reconstructions non souhaitées de votre objet, reportez-vous à 
+[Mon provider fait des mises à jour trop souvent, que puis-je faire ?](#my-provider-updates-too-often-what-can-i-do)
```
Comment (rrousselGit) [inline website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/combining_providers.mdx]:
```suggestion
[Mon provider fait des mises à jour trop souvent, que puis-je faire ?](#mon-provider-fait-des-mises-à-jour-trop-souvent-que-puis-je-faire-)
```

File: website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/combining_providers.mdx [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+```dart
+final myProvider = Provider((ref) {
+  // Mauvaise pratique d'appeler `read` ici.
+  final value = ref.read(anotherProvider);
+});
+```
+
+Si vous avez utilisé [read] pour tenter d'éviter les reconstructions non souhaitées de votre objet, reportez-vous à 
+[Mon provider fait des mises à jour trop souvent, que puis-je faire ?](#Mon-provider-fait-des-mises-à-jour-trop-souvent,-que-puis-je-faire-?)
```
Comment (rrousselGit) [inline website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/combining_providers.mdx:195]:
```suggestion
[Mon provider fait des mises à jour trop souvent, que puis-je faire ?](#mon-provider-fait-des-mises-à-jour-trop-souvent-que-puis-je-faire-)
```

**Comment by rrousselGit** [comment]
LGTM thanks!

**Pull request merged**